internal/plugin/plugin.go

```diff
@@ -32,59 +32,63 @@ var (
 )
 
 const (
-	ExtraPluginDirectoryEnvKey = "ATLAS_CLI_EXTRA_PLUGIN_DIRECTORY"
-	SourceType                 = "plugin"
+	PluginSourceType = "plugin"
+	sourceType       = "sourceType"
+	sourcePluginName = "sourcePluginName"
 )
 
 func IsPluginCmd(cmd *cobra.Command) bool {
-	if cmdSourceType, ok := cmd.Annotations["sourceType"]; ok && cmdSourceType == SourceType {
+	if cmdSourceType, ok := cmd.Annotations[sourceType]; ok && cmdSourceType == PluginSourceType {
 		return true
 	}
 	return false
 }
 
-func GetAllValidPlugins(existingCommands []*cobra.Command) []*Plugin {
-	var manifests []*Manifest
-
-	// Load manifests from plugin directories
-	if defaultPluginDirectory, err := GetDefaultPluginDirectory(); err == nil {
-		if loadedManifests, err := getManifestsFromPluginsDirectory(defaultPluginDirectory); err != nil {
-			logPluginWarning(`could not load manifests from directory "%s" because of error: %s`, defaultPluginDirectory, err.Error())
-		} else {
-			manifests = append(manifests, loadedManifests...)
-		}
+func GetPluginWithName(name string, existingCommandsMap map[string]bool, onlySearchValidPlugins bool) (*Plugin, error) {
+	var plugins []*Plugin
+	if onlySearchValidPlugins {
+		plugins = GetAllValidPlugins(existingCommandsMap)
+	} else {
+		plugins = getAllPlugins()
 	}
 
-	if extraPluginDir := os.Getenv("ATLAS_CLI_EXTRA_PLUGIN_DIRECTORY"); extraPluginDir != "" {
-		if loadedManifests, err := getManifestsFromPluginsDirectory(extraPluginDir); err != nil {
-			logPluginWarning(`could not load plugins from folder "%s" provided in environment variable ATLAS_CLI_EXTRA_PLUGIN_DIRECTORY: %s`, extraPluginDir, err.Error())
-		} else {
-			manifests = append(manifests, loadedManifests...)
+	for _, plugin := range plugins {
+		if plugin.Name == name {
+			return plugin, nil
 		}
 	}
 
+	return nil, fmt.Errorf("could not find plugin %s", name)
+}
+
+func getAllPlugins() []*Plugin {
+	// Load manifests from plugin directories
+	manifests := loadManifestsFromPluginDirectories()
+
+	// Convert manifests to plugins
+	plugins := convertManifestsToPlugins(manifests)
+
+	return plugins
+}
+
+func GetAllValidPlugins(existingCommandsMap map[string]bool) []*Plugin {
```

Comment:
Can you just pass a slice of strings here? (commands).

The map you're passing is basically:
```go
	s := map[string]bool{
		"command1": true,
		"command2": true,
		"command3": true,
		"command4": true,
		"command5": true,
	}
```

If you need a `map[string]bool` inside of this function you can always convert it inside of the function.